spring-cloud-task-stream/src/main/java/org/springframework/cloud/task/launcher/TaskLaunchRequest.java

```diff
@@ -90,6 +96,26 @@ public String getUri() {
 		return deploymentProperties;
 	}
 
+	/**
+	 * Returns the name that will be associated with the launched task.
+	 *
+	 * @return string containing the application name.
+	 */
+	public String getApplicationName() {
+		return applicationName;
+	}
+
+	/**
+	 * Sets the name to be applied to the launched task.   If set
+	 * 	to null then the launched task name will be "Task-<hash code of the task
+	 * 	launch request>.
+	 * @param applicationName the name to be
+	 */
+	public void setApplicationName(String applicationName) {
+		this.applicationName = applicationName == null ? "Task-" + hashCode() :
```

Comment:
Why are we using hash code here?  And it doesn't feel right to me to change the state of the object with the hash code like that.  Correct me if I'm wrong, but after calling this method, if the name is foo-123 (123 is my hash code), if I call `hashCode()` on this object now, I won't get 123 again.  If that's the case, we shouldn't advertise this as the object's hash code.  I'd take a random UUID personally...